service/src/main/java/uk/gov/hmcts/reform/fpl/service/CreateC21OrderService.java

```diff
@@ -0,0 +1,95 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.google.common.collect.ImmutableMap;
+import org.assertj.core.util.Lists;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.config.HmctsCourtLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.model.C21Order;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.common.C21OrderBundle;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.JudgeAndLegalAdvisor;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+import uk.gov.hmcts.reform.fpl.utils.JudgeAndLegalAdvisorHelper;
+
+import java.time.format.FormatStyle;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang3.ObjectUtils.defaultIfNull;
+
+@Service
+public class CreateC21OrderService {
+
+    private final DateFormatterService dateFormatterService;
+    private final HmctsCourtLookupConfiguration hmctsCourtLookupConfiguration;
+    private final Time time;
+
+    public CreateC21OrderService(DateFormatterService dateFormatterService,
+                                 HmctsCourtLookupConfiguration hmctsCourtLookupConfiguration, Time time) {
+        this.dateFormatterService = dateFormatterService;
+        this.hmctsCourtLookupConfiguration = hmctsCourtLookupConfiguration;
+        this.time = time;
+    }
+
+    public Map<String, Object> getC21OrderTemplateData(CaseData caseData) {
+        return ImmutableMap.<String, Object>builder()
+            .put("familyManCaseNumber", caseData.getFamilyManCaseNumber())
+            .put("courtName", getCourtName(caseData.getCaseLocalAuthority()))
+            .put("orderTitle", getOrderTitle(caseData))
+            .put("orderDetails", caseData.getTemporaryC21Order().getOrderDetails())
+            .put("todaysDate", dateFormatterService.formatLocalDateToString(time.now().toLocalDate(), FormatStyle.LONG))
+            .put("judgeTitleAndName", JudgeAndLegalAdvisorHelper.formatJudgeTitleAndName(
+                caseData.getJudgeAndLegalAdvisor()))
+            .put("legalAdvisorName", JudgeAndLegalAdvisorHelper.getLegalAdvisorName(
+                caseData.getJudgeAndLegalAdvisor()))
```

Comment:
hmm there is no consistency with how this was done in draft SDO 🤔

Reply:
To be honest I didn't fully see how it was done in draft SDO. I followed how it was done in Notice of Proceedings and used that as an example. Maybe needs a bigger discussion regarding consistency - a few of the comments on this PR are asking why I've done XYZ when really I've followed examples from existing code already in prod, assuming it's okay (maybe it was wrong to assume it's okay). I've used a lot of files for reference as I try and get my head around things but ultimately, if it's not okay now then why was it okay in a previous PR?

Reply:
I guess different people reviewing have different opinions etc. I guess this flags up another consistency / convention thing. Would be nice to iron this out. I like the idea of a JudgeAndLegalAdvisor service. This should be implemented everywhere when we can